backend/test/deal-observer.test.js

```diff
@@ -107,6 +107,36 @@ describe('deal-observer-backend', () => {
     await storeBlockEvent(data)
     assert.strictEqual(await countStoredActiveDeals(pgPool), 2n)
   })
+
+  it('deal with duplicate events', async () => {
+    const storeDeal = async (eventData) => {
+      const event = Value.Parse(BlockEvent, { height: 1, event: eventData, emitter: 'f06' })
+      const dbEntry = convertBlockEventToActiveDealDbEntry(event)
+      await storeActiveDeals([dbEntry], pgPool)
+    }
+    const eventData = {
+      id: 1,
+      provider: 2,
+      client: 3,
+      pieceCid: 'baga6ea4seaqc4z4432snwkztsadyx2rhoa6rx3wpfzu26365wvcwlb2wyhb5yfi',
+      pieceSize: 4n,
+      termStart: 5,
+      termMin: 12340,
+      termMax: 12340,
+      sector: 6n,
+      payload_cid: undefined,
+      payload_retrievability_state: PayloadRetrievabilityState.NotQueried,
+      last_payload_retrieval_attempt: undefined
+    }
+    await storeDeal(eventData)
+    let expected = await loadDeals(pgPool, 'SELECT * FROM active_deals')
+    assert.strictEqual(expected.length, 1)
+    // If we only change the id, the unique constraint which does not include the id should prevent the insertion
```

Comment:
```suggestion
    // If we only change the id, the unique constraint which does not include the id should not prevent the insertion
```

IIUC, in this test, we verify that the the unique constraint is not violated and the SQL query does not throw an error. As I understand the original comment, it describes the situation before we fixed it yesterday.

Reply:
The unique constraint does prevent the insertion of the duplicate. The handling of a conflict on unique constraint handles the conflict. At least that is how I understood it.

Reply:
Here it _does_ prevent the insertion of a new record, but it doesn't throw an error doing so. I think the comment is correct, but we could add about the error
```suggestion
    // If we only change the id, the unique constraint should prevent the insertion and fail silently
```

Reply:
@NikolasHaimerl you requested review again without resolving this thread, and there are others that are still unresolved. Could you please first resolve the ones that have already been completed, and address the ones that haven't?

Reply:
Right. I agree the comment correctly describes the unique constraint. I am confused about how the comment relates to what is being tested here.

Suggestion:

```suggestion
    // If we only change the id, the unique constraint which does not include the id will prevent the insertion
    // This test verifies that `storeDeal` handles such situation by ignoring the duplicate deal record
```

Reply:
+1 to will over should